testsuites/store/src/main/java/org/eclipse/rdf4j/sail/SailConcurrencyTest.java

```diff
@@ -127,7 +127,7 @@ public void run() {
 						conn.addStatement(subject, RDFS.COMMENT, vf.createLiteral("ci" + i), context);
 					}
 					if (rollback) {
-						otherTxnCommitted.await();
+						otherTxnCommitted.await(); // deadlocks here. Waiting for another transaction while holding the exclusive write lock!
 						conn.rollback();
```

Comment:
As an aside: when you commit changes to ignore this test, can you also remove the comment you added to the test case? 

Finally: release is only 2 days away. Can you finalize this PR by adding the required javadoc and making sure it's stable for merging?